bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -102,7 +103,24 @@ impl ClientBuilder {
             let data_path = PathBuf::from(base_path).join(sanitize(username));
             fs::create_dir_all(&data_path)?;
 
-            inner_builder = inner_builder.sled_store(data_path, builder.passphrase.as_deref());
+            let mut state_store =
+                matrix_sdk_sled::SledStateStore::builder().path(data_path.to_owned());
+
+            if let Some(passphrase) = builder.passphrase.as_deref() {
+                state_store = state_store.passphrase(passphrase.to_owned());
+            }
+
+            let state_store = state_store.build()?;
+
+            let crypto_store = RUNTIME.block_on(matrix_sdk_sqlite::SqliteCryptoStore::open(
```

Comment:
Maybe this method should be called something along the lines of `configureWithPath` if it does both creation and opening

Reply:
When you say creation, you mean creation of the DB or the `SqliteCryptoStore` object? 

If the former, you can't create a SQLite DB without opening it. If the later what's the point of having the object if it's unable to do anything except call a `build()` or `open()` method.

Note that the open method is somewhat idiomatic in the SQLite context: https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.open.

Reply:
> Note that the open method is somewhat idiomatic in the SQLite context

Right okay, that explains it